app/models/utils/nav/customization_parser.rb

```diff
@@ -5,10 +5,45 @@ module Nav
     # Parses the top menu customization setting
     class CustomizationParser
       attr_accessor :markdown
+      attr_accessor :customizations
 
       def initialize(markdown)
         # a nil input is stored as an empty string
         self.markdown = markdown || ""
+        self.customizations = parse
+        @done = []
+      end
+
+      def filter_item(name:, path:, permitted:, icon:)
+        # check to see if this item has been customized
+        result = {}
```

Comment:
I think if you just take a single argument in this method (e.g. `item`), it will get passed in as a hash, and then you can simplify a lot of the below since you don't have to re-assign name, path, permitted, etc. You just overwrite path if appropriate.

Reply:
yes, that works fine.